roadmap/implementers-guide/src/node/validity/assignments.md

```diff
@@ -0,0 +1,101 @@
+# Approval assignments
+
+Approval assignment determines on which candidate parachain blocks each validator performs approval checks.  An approval session consideres only one relay chain block and assigns only those candidates that relay chain block declares available.
+
+Assignment balances several concerns:
+
+- limits adversaries foreknowledge about assignments,
+- ensures enough checkers, and 
+- distributes assignments relatively equitably.  
+
+Assignees determine their own assignments to check specific candidates using two or three assignment criteria.  Assignees never reveal their assignments until relevant, and gossip delays assignments sent early, which limits others' foreknowledge.  Assignees learn their assignment only with the relay chain block. 
+
+All criteria require the validator evaluate a verifiable random function (VRF) using their VRF secret key.  All criteria input specific data called "stories" about the session's relay chain block, and output candidates to check and a precedence called a `DelayTranche`.  
+
+We liberate availability cores when their candidate becomes available of course, but one approval assignment criteria continues associating each candidate with the core number it occupied when it became available. 
+
+Assignment operates in loosely timed rounds determined by this `DelayTranche`s, which proceed roughly 12 times faster than six second block production assuming half second gossip times.  If a candidate `C` needs more approval checkers by the time we reach round `t` then any validators with an assignment to `C` in delay tranche `t` gossip their send assignment notice for `C`.  We continue until all candidates have enough approval checkers assigned.  We take entire tranches together if we do not yet have enough, so we expect strictly more than enough checkers.  We also take later tranches if some checkers return their approval votes too slow (see no shows below).  
```

Comment:
I wonder if this leans too heavily on the half-second gossip requirement.

How many `DelayTranches` do we run per relay-chain block?

What should validators do who are offline for some amount of time and are re-syncing the chain?

Reply:
We want delay tranches to run at about our gossip speed, while no show timeouts should be long enough to reconstruct and check, and presumably longer than one relay chain block. 

If they're resyncing then they should not send out assignment notices or do approval checks, which should enable them to resync the chain much faster than if they're validating those 6+ candidates per relay chain block .

Reply:
d3413e3e